src/WebJobs.Script/Description/DotNet/DotNetCompilationServiceFactory.cs

```diff
@@ -24,14 +27,31 @@ ImmutableArray<ScriptType> ICompilationServiceFactory.SupportedScriptTypes
             }
         }
 
+        private static OptimizationLevel OptimizationLevel
+        {
+            get
+            {
+                if (_optimizationLevel == null)
+                {
+                    string releaseModeSetting = Environment.GetEnvironmentVariable(DotNetConstants.CompilationReleaseMode);
+
+                    _optimizationLevel = string.Equals(releaseModeSetting, bool.TrueString, StringComparison.OrdinalIgnoreCase)
```

Comment:
So the app setting is meant to be a bool? Would there ever be a third option? I had expected you to be parsing `Release` or `Debug` here and it confused me for a sec.